pkg/bucket/bucket.go

```diff
@@ -248,25 +248,24 @@ func prepareQuotaConfig(bucketName string, maxSize string, maxObjects string) (n
 	quota := nb.QuotaConfig{}
 
 	// validate quota config for bucket
-	err := util.ValidateQuotaConfig(bucketName, maxSize, maxObjects)
-	if err != nil {
+	if err := util.ValidateQuotaConfig(bucketName, maxSize, maxObjects); err != nil {
 		return quota, false, err
 	}
 
 	if maxSize != "" {
 		quantity, _ := resource.ParseQuantity(maxSize)
 		bucketMaxSize = quantity.Value()
+		if bucketMaxSize > 0 {
+			f, u := nb.GetBytesAndUnits(bucketMaxSize, 2)
+			quota.Size = &nb.SizeQuotaConfig{Value: f, Unit: u}
+		}
 	}
 	if maxObjects != "" {
 		bucketMaxObjects, _ = strconv.ParseInt(maxObjects, 10, 64)
+		if bucketMaxObjects > 0 {
+			quota.Quantity = &nb.QuantityQuotaConfig{Value: int(bucketMaxObjects)}
+		}
 	}
 
-	if bucketMaxSize > 0 {
-		f, u := nb.GetBytesAndUnits(bucketMaxSize, 2)
-		quota.Size = &nb.SizeQuotaConfig{Value: f, Unit: u}
-	}
-	if bucketMaxObjects > 0 {
-		quota.Quantity = &nb.QuantityQuotaConfig{Value: int(bucketMaxObjects)}
-	}
-	return quota, true, err
+	return quota, true, nil
```

Comment:
Am I checking the right commit ? Still can see 3 values

Reply:
Updated the code. Thanks